test/runtests.jl

```diff
@@ -8,257 +8,314 @@ end
 const glm_datadir = joinpath(dirname(@__FILE__), "..", "data")
 
 ## Formaldehyde data from the R Datasets package
-form = DataFrame(Carb=[0.1,0.3,0.5,0.6,0.7,0.9],OptDen=[0.086,0.269,0.446,0.538,0.626,0.782])
-lm1 = fit(LinearModel, OptDen ~ Carb, form)
-test_show(lm1)
-@test_approx_eq coef(lm1) collect(linreg(convert(Array, form[:Carb]), convert(Array, form[:OptDen])))
-Σ = [6.136653061224592e-05 -9.464489795918525e-05
-    -9.464489795918525e-05 1.831836734693908e-04]
-@test_approx_eq vcov(lm1) Σ
-@test_approx_eq cor(lm1.model) diagm(diag(Σ))^(-1/2)*Σ*diagm(diag(Σ))^(-1/2)
-@test df(lm1) == 3
-@test_approx_eq deviance(lm1) 0.0002992000000000012
-@test_approx_eq loglikelihood(lm1) 21.204842144047973
-@test_approx_eq nulldeviance(lm1) 0.3138488333333334
-@test_approx_eq nullloglikelihood(lm1) 0.33817870295676444
-@test r²(lm1) == r2(lm1)
-@test_approx_eq r²(lm1) 0.9990466748057584
-@test adjr²(lm1) == adjr2(lm1)
-@test_approx_eq adjr²(lm1) 0.998808343507198
-@test_approx_eq aic(lm1) -36.409684288095946
-@test_approx_eq aicc(lm1) -24.409684288095946
-@test_approx_eq bic(lm1) -37.03440588041178
-
-dobson = DataFrame(Counts=[18.,17,15,20,10,20,25,13,12], Outcome=gl(3,1,9), Treatment=gl(3,3))
-gm1 = fit(GeneralizedLinearModel, Counts ~ Outcome + Treatment, dobson, Poisson())
-test_show(gm1)
-@test df(gm1) == 5
-@test_approx_eq deviance(gm1) 5.12914107700115
-@test_approx_eq loglikelihood(gm1) -23.380659200978837
-@test_approx_eq aic(gm1) 56.76131840195767
-@test_approx_eq aicc(gm1) 76.76131840195768
-@test_approx_eq bic(gm1) 57.74744128863877
-@test_approx_eq coef(gm1)[1:3] [3.044522437723423,-0.45425527227759555,-0.29298712468147375]
+form = DataFrame(Any[[0.1,0.3,0.5,0.6,0.7,0.9],[0.086,0.269,0.446,0.538,0.626,0.782]],
+    [:Carb, :OptDen])
+
+@testset "lm" begin
+    lm1 = fit(LinearModel, OptDen ~ Carb, form)
+    test_show(lm1)
+    @test isapprox(coef(lm1), collect(linreg(form[:Carb], form[:OptDen])))
```

Comment:
The reason for using `isapprox` is because the optional `atol` and `rtol` arguments are available. Sometimes the same calculation on different platforms can produce slightly different results and tolerances are helpful.